independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ObserverGenerator.java

```diff
@@ -414,8 +413,8 @@ protected void implementNotify(ObserverInfo observer, ClassCreator observerCreat
                 ResultHandle context = notify.invokeInterfaceMethod(MethodDescriptors.ARC_CONTAINER_GET_ACTIVE_CONTEXT,
                         container,
                         scope);
-                notify.ifNull(context).trueBranch().throwException(ContextNotActiveException.class,
-                        "Context not active: " + observer.getDeclaringBean().getScope().getDotName());
+                // if the context isn't active, don't notify the observer
+                notify.ifNull(context).trueBranch().returnVoid();
```

Comment:
I'm probably just being stupid here, but conditional observers (`Reception.IF_EXISTS`) are handled above (in the exact same way), so this case only applies to "regular" obsevers (`Reception.ALWAYS`). Why do we need this change? Not saying we don't, just want to understand.

Reply:
Okay, I see that this probably fixes `ObserverNotificationTest.testObserverMethodNotInvokedIfNoActiveContext`.